src/python/pants/backend/native/tasks/native_task.py

```diff
@@ -80,14 +85,12 @@ def _native_toolchain(self):
   def _toolchain_variant_request(self):
     return ToolchainVariantRequest(
       toolchain=self._native_toolchain,
-      variant=self._native_build_settings.toolchain_variant)
+      variant=self._native_build_step_settings.toolchain_variant)
 
-  @memoized_method
-  def get_c_toolchain_variant(self):
+  def get_c_toolchain_variant(self, native_library_target):
     return self._request_single(CToolchain, self._toolchain_variant_request)
 
-  @memoized_method
-  def get_cpp_toolchain_variant(self):
+  def get_cpp_toolchain_variant(self, native_library_target):
     return self._request_single(CppToolchain, self._toolchain_variant_request)
```

Comment:
@cosmicexplorer for `get_c{pp}_toolchain_variant`, should I take `native_library_target.payload.toolchain_variant` if exists, over `self._request_single(CppToolchain, self._toolchain_variant_request)`?

Reply:
Definitely let `native_library_target.payload.toolchain_variant` if specified -- kind of like #7105!